async_test.go

```diff
@@ -41,18 +42,28 @@ func validMongoDBDownloadOptions() MongoDBDownloadOptions {
 }
 
 func TestSetupDownloadMongoDBReleasesWithBadPath(t *testing.T) {
+	ctx, cancel := context.WithTimeout(context.Background(), taskTimeout)
+	defer cancel()
+
 	opts := validMongoDBDownloadOptions()
 	opts.Path = "async_test.go"
-	assert.Error(t, SetupDownloadMongoDBReleases(context.Background(), lru.NewCache(), opts))
+	assert.Error(t, SetupDownloadMongoDBReleases(ctx, lru.NewCache(), opts))
 }
 
 func TestSetupDownloadMongoDBReleasesWithBadArtifactsFeed(t *testing.T) {
+	ctx, cancel := context.WithTimeout(context.Background(), taskTimeout)
+	defer cancel()
+
 	opts := validMongoDBDownloadOptions()
 	opts.Path = filepath.Join("build", "full.json")
-	assert.Error(t, SetupDownloadMongoDBReleases(context.Background(), lru.NewCache(), opts))
+	err := SetupDownloadMongoDBReleases(ctx, lru.NewCache(), opts)
+	assert.Error(t, err)
 }
 
 func TestCreateValidDownloadJobs(t *testing.T) {
+	_, cancel := context.WithTimeout(context.Background(), taskTimeout)
+	defer cancel()
```

Comment:
use the context or omit this. I don't think it does anything.

Reply:
Removed.